broke-broker/src/main/java/org/dsngroup/broke/broker/storage/SubscriptionPool.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) 2017 original authors and authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dsngroup.broke.broker.storage;
+
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelHandlerContext;
+import org.dsngroup.broke.protocol.*;
+
+import java.util.*;
+
+/**
+ * Subscription pool that stores all subscriptions of a server session.
+ * */
+public class SubscriptionPool {
+
+    /**
+     * The subscriber pool
+     * The List of subscribers
+     * */
+    private List<Subscription> subscriptionList;
+
+    /**
+     *
+     * */
+    private Random packetIdRandomGenerator = new Random();
```

Comment:
Use UUID

Reply:
Since the packetId needs to be integer, a 32-byte UUID may not be a good choice. In [this](https://stackoverflow.com/questions/5563502/uuid-to-unique-integer-id) website, the static AtomicInteger is mentioned to be able to generate random integer at the runtime.

Reply:
Ok, just ensure there’s no conflict occurred.